### PR TITLE
Document CSRF middleware.

### PR DESCRIPTION
**Pull request by markstory**
Refs cakephp/cakephp#10561

### CODE REVIEW

File: en/controllers/middleware.rst [hunk 1]
```diff
@@ -322,6 +324,59 @@ backwards compatible with ``CookieComponent`` from earlier versions of CakePHP.
...
+CSRF Protection and AJAX Requests
+----------------------------------
+
+In addition to request data parameters, CSRF tokens can be submitted through
+a special ``X-CSRF-Token`` header. Using a header often makes it easier to
+integrate a CSRF token with JavaScript heavy applications, or XML/JSON based API
+endpoints.
+
+.. versionadded:: 3.5.0
+    The ``CsrfProtectionMiddleware`` was added in 3.5.0
```
Comment (ravage84) [inline en/controllers/middleware.rst]: Shouldn't this note be under "Cross Site Request Forgery (CSRF) Middleware"?

Comment (markstory) [inline en/controllers/middleware.rst]: Yes it should.

File: en/controllers/middleware.rst [hunk 2]
```diff
@@ -322,6 +324,63 @@ backwards compatible with ``CookieComponent`` from earlier versions of CakePHP.
...
+
+    When using CSRF protection you should always start your forms with the
+    FormHelper. If you do not, you will need to manually create hidden inputs in
+    each of your forms.
+
+CSRF Protection and AJAX Requests
+----------------------------------
+
+In addition to request data parameters, CSRF tokens can be submitted through
+a special ``X-CSRF-Token`` header. Using a header often makes it easier to
```
Comment (inoas) [inline en/controllers/middleware.rst]: Point to docs how to do it?
Is it enabled by default?
Should it?

Comment (markstory) [inline en/controllers/middleware.rst]: Setting headers is part of whatever client library.

**Pull request merged**